demos/python_demos/multi_camera_multi_person_tracking/utils/network_wrappers.py

```diff
@@ -81,3 +105,215 @@ def forward(self, batch):
+                    continue
+                bbox = [int(value) for value in boxes[i]]
+                if return_cropped_masks:
+                    left, top, right, bottom = bbox
+                    mask = masks[i][top:bottom, left:right]
+                else:
+                    mask = masks[i]
+                frame_output.append([bbox, scores[i], mask])
+            outputs.append(frame_output)
+        return outputs
+
+    def run_asynch(self, frames, index):
+        self.frames = frames
+
+    def wait_and_grab(self):
+        return self.get_detections(self.frames)
+
+    class Compose(object):
+        def __init__(self, transforms):
+            self.transforms = transforms
+
+        def __call__(self, sample):
+            for t in self.transforms:
+                sample = t(sample)
+            return sample
+
+    class Resize(object):
+        def __init__(self, max_size=None, window_size=None, size=None):
+            super().__init__()
+            assert int(max_size is not None) + int(window_size is not None) + int(size is not None) == 1
+
+            self.short_side_max = None
+            self.long_side_max = None
+            if max_size is not None:
+                self.short_side_max, self.long_side_max = max_size
+
+            self.height_max = None
+            self.width_max = None
+            if window_size is not None:
+                self.height_max, self.width_max = window_size
+
+            self.height = None
+            self.width = None
+            if size is not None:
+                self.height, self.width = size
+
+        def get_scale(self, image_size):
+            if self.height is not None:
+                scale_x, scale_y = self.width / image_size[1], self.height / image_size[0]
+            elif self.height_max is not None:
+                im_scale = min(self.height_max / image_size[0], self.width_max / image_size[1])
+                scale_x, scale_y = im_scale, im_scale
+            else:
+                im_scale = min(self.short_side_max / min(image_size), self.long_side_max / max(image_size))
+                scale_x, scale_y = im_scale, im_scale
+            return scale_x, scale_y
+
+        def __call__(self, sample):
+            image_size = sample['image'].shape[:2]
+            scale_x, scale_y = self.get_scale(image_size)
+
+            # Resize image.
+            sample['image'] = cv2.resize(sample['image'], None, fx=scale_x, fy=scale_y)
+            h, w = sample['image'].shape[:2]
+
+            # Resize boxes.
+            if 'gt_boxes' in sample:
+                sample['gt_boxes'] *= [scale_x, scale_y, scale_x, scale_y]
+                sample['gt_boxes'] = np.clip(sample['gt_boxes'], 0, [w - 1, h - 1, w - 1, h - 1])
+
+            # Resize masks.
+            if 'gt_masks' in sample:
+                sample['gt_masks'] = [[np.clip(part * [scale_x, scale_y], 0, [w - 1, h - 1]) for part in obj]
+                                      for obj in sample['gt_masks']]
+
```

Comment:
`gt_boxes` and `gt_masks` paths seem to be never used. The whole class can be much sorter:
```py
scale = min(net_w / image_size[0], net_h / image_size[1])
sample['image'] = cv2.resize(sample['image'], None, fx=scale, fy=scale)
```
Since `MaskRCNN` already stores that `net_w` and `net_h` as `w` and `h`, there is no reason to introduce `Resize` class (if `Normalize` is removed and `Compes` becomes unusd), it should be a function (if it is worth it to introduce a function containing only 2 lines and which is used only in one place).

Reply:
Cleaned.